CMakeLists.txt

```
@@ -886,6 +886,16 @@ cmake_dependent_option(
   Will be disabled if not supported by the platform" ON
   "USE_CUDA OR USE_ROCM" OFF)
 
+#
+# Cannot be put into Dependencies.cmake due circular dependency:
+# USE_FLASH_ATTENTION -> USE_ROCM -> Dependencies.cmake -> aotriton.cmake
+#
+if(USE_ROCM)
+  if(USE_FLASH_ATTENTION OR USE_MEM_EFF_ATTENTION)
+    include(cmake/External/aotriton.cmake)
+  endif()
+endif()
```

Can't one have composite statements in CMake?
```suggestion
if(USE_ROCM AND (USE_FLASH_ATTENTION OR USE_MEM_EFF_ATTENTION))
  include(cmake/External/aotriton.cmake)
endif()
```

---

It is possible. However, the quadruple roles of `USE_XXX` variable means it is likely that more logic will have to be deferred to the main CMakeLists.txt. (As far as I'm aware of, Flash Attention implemented with CK backend also fits this situation.)

A dedicated `if(USE_ROCM)-endif()` will make it more readable when more things have to be moved here.